src/main/java/com/github/lmiljak/dowertefence/OnCommandGameCreator.java

```diff
@@ -0,0 +1,276 @@
+					return false;
+				}
+			}
+
+			return true;
+		}
+
+	}
+
+	/**
+	 * Constructor for OnCommandGameCreator.
+	 */
+	public OnCommandGameCreator() {
+		this.listeners = new HashSet<>();
+		this.potentialGames = new HashMap<>();
+	}
+
+	@Override
+	public boolean onCommand(CommandSender sender, Command command, String label, String[] args) {
+		// Processes the start and accept sub commands
+
+		if (!(sender instanceof Player)) {
+			sender.sendMessage("You must be a Player to send this message");
+			return false;
+		}
+		if (args.length == 0) {
+			// There should always be at least one argument such as "start" or
+			// "accept".
+			return false;
+		}
+
+		if (label.equals(DowerTefence.PARENT_COMMAND)) {
+			switch (args[0]) {
+			case "start":
+				return processStartCommand((Player) sender, removeFirst(args));
+			case "accept":
+				return processAcceptCommand((Player) sender, removeFirst(args));
+			default:
+				return false;
+			}
+		}
+
+		return false;
+	}
+
+	/**
+	 * Takes a String array and removes the first String from it.
+	 * 
+	 * @param stringArray
+	 *            The String array to remove the first element of.
+	 * @return The new String array.
+	 */
+	private String[] removeFirst(String[] stringArray) {
+		String[] result = new String[stringArray.length - 1];
+
+		for (int i = 1; i < stringArray.length; i++) {
+			result[i - 1] = stringArray[i];
+		}
+
+		return result;
+	}
+
+	/**
+	 * Used to process the start command. If the sender of the command invited
+	 * no people, the game is immediately created. In the case that the sender
+	 * did invite people, a PotentialGame will be created until all players have
+	 * accepted.
+	 * 
+	 * @param sender
+	 *            The player that sent the command.
+	 * @param invitedPlayers
+	 *            The players that got invited by sender (the arguments of the
+	 *            command).
+	 * @return True if and only if the command was executed successfully.
+	 */
+	private boolean processStartCommand(Player sender, String[] invitedPlayers) {
+		HashSet<Player> participants = new HashSet<>();
+
+		// Check if the sender wants to play single player, if so then just
+		// create the game with him in it
+		if (invitedPlayers.length == 0) {
+			participants.add(sender);
+			notifyListeners(new StandardGame(participants));
+			return true;
+		}
+
+		participants = invitePlayers(sender, invitedPlayers);
+		if (participants == null) {
+			// The invites have been cancelled and the sender got notified.
+			return true;
+		}
+
+		for (String arg : invitedPlayers) {
+			Player invitedPlayer = Bukkit.getPlayer(arg);
+
+			if (invitedPlayer == null) {
+				sender.sendMessage("One or more of the specified players do not exist or are not online.");
+				return true;
+			}
+
+			participants.add(invitedPlayer);
+		}
+
+		// Creating a new potential game.
+		potentialGames.put(sender, new PotentialGame(participants));
+		sender.sendMessage("Waiting for all invited players to accept.");
```

Comment:
This reinforces my believe that everyone needs to accept for a game to start. People should also be able to reject/deny and be removed from the potentialPlayers map.